cpp/conversation.h

```diff
@@ -0,0 +1,163 @@
+/*!
+ *  Copyright (c) 2023 by Contributors
+ * \file llm_chat.cc
+ * \brief Implementation of llm chat.
+ */
+#define PICOJSON_USE_INT64
+#define __STDC_FORMAT_MACROS
+#include <tvm/runtime/module.h>
+
+#include <string>
+#include <vector>
+
+namespace mlc {
+namespace llm {
+
+enum class SeparatorStyle {
+  kAddColon,
+  kLM,
+};
+
+/*!
+ * \brief helper class to keep track of conversation.
+ */
+class Conversation {
```

Comment:
Shall we make it TVM runtime object so that we can use it in other languages (e.g. Python) as well?

Reply:
this is a good pt, we could do that as a folllowup